core/src/main/java/org/ldaptive/ext/MergeOperation.java

```diff
@@ -402,4 +445,22 @@ protected Result delete(final MergeRequest request, final LdapEntry entry)
     logger.info("Delete entry {} for request {}", entry, request);
     return result;
   }
+
+
+  /**
+   * Divides the supplied list into sub lists by the supplied divisor and passes each sub list to the consumer.
+   *
+   * @param  <T>  type of list element
+   * @param  list  to divide
+   * @param  divisor  to divide list by
+   * @param  consumer  to process each sub list
+   */
+  private <T> void divideList(final List<T> list, final int divisor, final Consumer<List<T>> consumer)
```

Comment:
I wonder whether a `Spliterator` may be more performant here. The usage example at https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Spliterator.html strikes me as awfully similar, and may provide an opportunity for parallelism. Moreover, a dedicated implementation of `Spliterator` may provide a dedicated component to deal with batching concerns. The changes in this MR add a fair bit of complexity to this class soley for batching, and they may be better organized in a separate class.

Reply:
That's a great suggestion, I'll investigate.

Reply:
Pushed a commit to reduce complexity in `MergeOperation` by adding a new interface. I didn't get to a `Spliterator` yet, but I think this is moving in the right direction.

Reply:
Agree, it's a step in the right direction.